### PR TITLE
Implement `collect()` function for aggregating values into lists

### PR DESCRIPTION
**Pull request by amansinghoriginal**
# Description

This PR implements the collect() function for Drasi Core which enables collecting values into lists during query execution. This is a common aggregation function in Cypher that allows for creating structured results with nested collections.

Fixes: #43 

## Type of change
- [ ] This pull request fixes a bug in Drasi and has an approved issue (issue link required).
- [x] This pull request adds or changes features of Drasi and has an approved issue (issue link required).
- [ ] This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Drasi (issue link optional).

## Features
- Added implementation of the collect() aggregation function
- Support for collecting any value type (primitives, maps, etc.) into lists
- Proper handling of additions, updates, and deletions of collection elements
- Support for null value handling (nulls are ignored in collection)
- Implemented efficient state tracking with a count-based accumulator

## Implementation Details
The implementation tracks collection elements efficiently using a counting approach:

- Uses a HashMap to store unique values and their counts
- Properly handles duplicates by maintaining count information
- Implements revert operations to correctly remove elements when needed
- Emits appropriate diffs when collections change

## Testing
- Added unit tests covering basic functionality, duplicates, nulls, and edge cases
- Added a comprehensive end-to-end test with a movie database scenario

### CODE REVIEW

File: core/src/evaluation/functions/.DS_Store
Comment (danielgerlag) [inline core/src/evaluation/functions/.DS_Store:1]: We should gitignore this file

File: core/src/evaluation/functions/aggregation/mod.rs [hunk 1]
```diff
@@ -80,6 +84,13 @@ pub enum ValueAccumulator {
         cov: f64,
     },
     Value(ElementValue),
+    CollectMultiset {
```
Comment (danielgerlag) [inline core/src/evaluation/functions/aggregation/mod.rs:87]: You would need to update the RocksDb and Redis storage models to support these new variants, this will only work for "In Memory"

File: core/src/evaluation/functions/aggregation/mod.rs [hunk 2]
```diff
@@ -80,6 +84,13 @@ pub enum ValueAccumulator {
...
+        list: Vec<ElementValue>,
```
Comment (danielgerlag) [inline core/src/evaluation/functions/aggregation/mod.rs:88]: What would the trade offs be of just using `ValueAccumulator::Value(ElementValue::List)`, instead of these new variants?

File: core/src/evaluation/functions/aggregation/mod.rs [hunk 3]
```diff
@@ -80,6 +84,13 @@ pub enum ValueAccumulator {
...
+        counts: std::collections::HashMap<ElementValue, usize>,
+    },
+    CollectCounts {
```
Comment (danielgerlag) [inline core/src/evaluation/functions/aggregation/mod.rs:91]: What are the purposes / differences of each of these new accumulator variants?

File: shared-tests/src/in_memory/mod.rs [hunk 4]
```diff
@@ -122,6 +122,17 @@ mod remap {
     }
 }
 
+mod collect_movies {
+    use super::InMemoryQueryConfig;
+    use crate::use_cases::*;
+
+    #[tokio::test]
+    pub async fn collect_movies() {
```
Comment (danielgerlag) [inline shared-tests/src/in_memory/mod.rs:130]: We should surface this test in the Redis & RocksDb tests as well.
